integrationtests/tests/fixtures/mod.rs

```diff
@@ -945,16 +958,18 @@ impl FederationTest {
     /// it will wait forever
     pub async fn await_consensus_epochs(&self, epochs: usize) -> anyhow::Result<()> {
         for _ in 0..(epochs) {
-            for maybe_cancelled in join_all(
-                self.servers
-                    .iter()
-                    .map(|server| Self::consensus_epoch(server.clone(), Duration::from_millis(0))),
-            )
-            .await
-            {
-                maybe_cancelled?;
+            let mut task_group = TaskGroup::new();
+            for (i, server) in self.servers.iter().enumerate() {
+                let server = server.clone();
+                task_group
+                    .spawn(format!("server-{i}-consensu_epoch"), move |_| async {
+                        Self::consensus_epoch(server, Duration::from_millis(0)).await
```

Comment:
This was the main goal here. `await_consensus_epochs` is used in a serial fashion all over the test suite and has a significant impact.